vm/option.go

```diff
@@ -50,44 +50,44 @@ func NewOption[T any](namespace string, defaultConfig T, optionFunc OptionFunc[T
 	return newOptionWithBytes(namespace, configOptionFunc)
 }
 
-func WithBuilder() RegisterFunc {
-	return func(vm *VM) {
-		vm.builder = builder.NewManual(vm)
+func WithBuilder[T chain.RuntimeInterface]() RegisterFunc[T] {
+	return func(vm *VM[T]) {
+		vm.builder = builder.NewManual[T](vm)
 	}
 }
 
-func WithGossiper() RegisterFunc {
-	return func(vm *VM) {
-		vm.gossiper = gossiper.NewManual(vm)
+func WithGossiper[T chain.RuntimeInterface]() RegisterFunc[T] {
+	return func(vm *VM[T]) {
+		vm.gossiper = gossiper.NewManual[T](vm)
 	}
 }
 
-func WithManual() Option {
-	return NewOption[struct{}](
+func WithManual[T chain.RuntimeInterface]() Option[T] {
+	return NewOption(
 		"manual",
 		struct{}{},
-		func(vm *VM, _ struct{}) error {
-			WithBuilder()(vm)
-			WithGossiper()(vm)
+		func(vm *VM[T], _ struct{}) error {
+			WithBuilder[T]()(vm)
+			WithGossiper[T]()(vm)
 			return nil
 		},
 	)
 }
 
-func WithBlockSubscriptions(subscriptions ...event.SubscriptionFactory[*chain.StatefulBlock]) RegisterFunc {
-	return func(vm *VM) {
+func WithBlockSubscriptions[T chain.RuntimeInterface](subscriptions ...event.SubscriptionFactory[*chain.StatefulBlock[T]]) RegisterFunc[T] {
+	return func(vm *VM[T]) {
 		vm.blockSubscriptionFactories = append(vm.blockSubscriptionFactories, subscriptions...)
 	}
 }
 
-func WithVMAPIs(apiHandlerFactories ...api.HandlerFactory[api.VM]) RegisterFunc {
-	return func(vm *VM) {
+func WithVMAPIs[T chain.RuntimeInterface](apiHandlerFactories ...api.HandlerFactory[api.VM[T]]) RegisterFunc[T] {
+	return func(vm *VM[T]) {
 		vm.vmAPIHandlerFactories = append(vm.vmAPIHandlerFactories, apiHandlerFactories...)
 	}
 }
 
-func WithTxRemovedSubscriptions(subscriptions ...event.SubscriptionFactory[TxRemovedEvent]) RegisterFunc {
-	return func(vm *VM) {
+func WithTxRemovedSubscriptions[T chain.RuntimeInterface](subscriptions ...event.SubscriptionFactory[TxRemovedEvent]) RegisterFunc[T] {
```

Comment:
For this PR my thought was that we would just make the runtime type a required parameter for hypersdk's new ([ref](https://github.com/ava-labs/hypersdk/blob/370360d6c7a2b9a5f3b9ffcdce63356b664e21f1/vm/vm.go#L150)). I don't think we can do an option like I previously thought since the `VM` is parameterized on a generic type so the caller must define it.